packages/react/src/use-chat.ts

```diff
@@ -533,6 +576,7 @@ By default, it's set to 0, which will disable the feature.
               id: generateId(),
               role: 'user',
               content: input,
+              experimental_attachments: attachmentsForRequest,
```

Comment:
will this add an empty array in many cases (increasing msg size)?

Reply:
it won't increase the message size, it gets filtered out when constructing the request payload [here](https://github.com/vercel/ai/pull/2226/files#diff-0a57b9a56351ecd277e650d5a02072fbb65c6d685b81103e1cdf0459cf7c323eR123-R125)

Reply:
how would it get filtered there? wouldn't it still add an empty array? maybe I'm missing something

Reply:
if `experimental_attachments` is not used in `handleSubmit`, then it will remain `undefined` and get filtered out in the payload

Reply:
but it's set to `[]` in line 535

Reply:
ah you're right! updated to add a check to set it as `undefined` if the array is empty